### ISSUE TITLE
[BUG] potential bug: 

### ISSUE THREAD

**Issue by lenaklosik** [sktime/sktime#7454]
**Potential Bug Description**
There is an inconsistency in the lagging behavior of `y` variable between the `RecursiveReductionForecaster` (RRF) and the `DirectReductionForecaster` (DRF).

In the `RecursiveReductionForecaster` (RRF):
An additional automatic lag of 1 is applied to y on top of the user-specified window_length.
This results in lags ranging from 1 to window_length.

Example
- window_length = 12
- lagged y (Xtt): lag1_lag0, lag1_lag1, ..., lag1_lag11
_(Lags range from 1 to 12)_

In the `DirectReductionForecaster` (DRF):
The extra lag is absent, and the lags range directly from 0 to window_length - 1.

Example
- window_length = 12
- lagged y (Xtt): lag0_lag0, lag0_lag1, ..., lag0_lag11 
_(Lags range from 0 to 11)_

**Impact:**
This inconsistency causes a different interpretation of the window_length parameter between the two forecasters.
It may lead to unexpected behavior when transitioning between RecursiveReductionForecaster and DirectReductionForecaster.

Is this behavior intentional to account for specific design considerations, or is it an oversight?

See discussion in discord:
https://discord.com/channels/1075852648688930887/1308693497263423509

**Comment by Sohaib-Ahmed21** [comment]
@lenaklosik I've researched upon it, `DirectReductionForecaster` is lagging `y`(target of estimator) unlike `RecursiveReductionForecaster`. Due to this, when features are made, there is no inconsistent behavior and everything seems fine regarding `window_length `and `lag `handling. Inner logic handles it all and features are in accordance with what described here https://gist.github.com/mlisovyi/f9944778da6d9ee12626413602543629#recursivereductionforecaster
@fkiraly your thoughts will be helpful.

**Comment by fkiraly** [comment]
@Sohaib-Ahmed21, can you kindly provide more detail on what the inner features are in which cases? Plus whether you think they are correct or not.

**Comment by Sohaib-Ahmed21** [comment]
> Are you saying:
> 
> * the relative shifts in both direct and recursive cases are correct (as expected per docstring)
> * but in one case we shift the features by one, and in one case we shift the targetr by -1?

Yes, exactly that's the case and due to this internal handling of logic, features are correctly produced without data leakage.

> Further, I notice that applying to laggers leads to funny column names. We should probably change that to the correct relative 
> shift, as it is very obscure to a user. I would even consider the column naming in the recursive forecaster a bug.

Yup, names seem to be obscure, probably we can avoid naming like **lag_1_lag_0_T,  lag_1_lag_1_T**,etc. by setting `keep_column_names=True` while applying **extra lag of 1** as I have done in this PR https://github.com/sktime/sktime/pull/7678. Your thoughts will be helpful.
